sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc

```diff
@@ -0,0 +1,220 @@
+
+* load/store flexible amount of elements per work item as sycl::span
+
+* specify data placement type via properties (1) or group_memory_helper (*naming is TBD) class (5)
+
+* extra options such as setting boundry values or limiting numbers of work items can be also specified via group_memory_helper
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Load API
+template<typename GroupHelper, typename InputIterator, typename OutputT,
+        std::size_t ElementsPerWorkItem, typename Properties = properties<>>
+void load_over_group(GroupHelper gh, InputIteratorT in_ptr,
+                      sycl::span<OutputT, ItemsPerWorkItem> out, Properties = {}); (1)
+
+template<typename GroupHelper, typename InputIterator, typename OutputT,
+        std::size_t ElementsPerWorkItem, typename GroupMemoryHelper>
+void load_over_group(GroupHelper gh, InputIteratorT in_ptr,
+                      sycl::span<OutputT, ItemsPerWorkItem> out,
+                      GroupMemoryHelper<typename InputIterator::value_type, ItemsPerWorkItem> memory_helper = {}); (2)
+
+// Store API
+template<typename GroupHelper, typename OutputIterator, typename InputT,
+        std::size_t ElementsPerWorkItem, typename Properties = properties<>>
+void store_over_group(GroupHelper gh, OutputIterator out_ptr,
+                      sycl::span<InputT, ItemsPerWorkItem> in, Properties = {}); (3)
+
+template<typename GroupHelper, typename OutputIterator, typename InputT,
+                  std::size_t ElementsPerWorkItem, typename GroupMemoryHelper>
+void store_over_group(GroupHelper gh, OutputIterator out_ptr,
+                      sycl::span<InputT, ItemsPerWorkItem> in,
+                      GroupMemoryHelper<InputT, ItemsPerWorkItem> memory_helper = {}); (4)
+
+// Properties:
+enum class group_algorithm_data_placement {
+  blocked,
+  striped
+};
+
+// Group Memory Helper (aka Loader/Storer) TBD:
+template<typename T, std::size_t ElementsPerWorkItem>
+class group_memory_helper {
+public:
+  template<typename Properties>
+  group_memory_helper(Properties properties = {});
+
+  void set_valid_items_num(std::size_t valid_items);
+
+  void set_out_of_boundry_value(T oob_default);
+
+  static constexpr std::size_t memory_required(sycl::memory_scope scope, std::size_t block_size);
+}; (5)
+
+}
+----
+
+(1) _Preconditions_: `in_ptr` must be the same for all work-items
+in the group.
+
+_Constraints_: Only available if `GroupHelper` was created with a
+work-group or sub-group and some associated scratch space.
+
+_Effects_: Loads `ItemsPerWorkItem` elements from `in_ptr` to `out`
+using the `gh` group helper object. Properties can specify data placement.
+
+
+(2) _Preconditions_: `in_ptr` must be the same for all work-items
+in the group.
```

Comment:
Should there be a precondition that `out` must be a pointer to the private address space?

To my mind, this restriction is what makes this a "load" and not a "copy".  If `out` can be a pointer to global memory, the function would be able to "load" from global memory and implicitly store the result back to global memory, which would be weird.